api/v1beta1/common.go

```diff
@@ -39,3 +39,12 @@ type GrafanaCommonSpec struct {
 	// +optional
 	AllowCrossNamespaceImport *bool `json:"allowCrossNamespaceImport,omitempty"`
 }
+
+// Common Functions that all CRs should implement, excluding Grafana
+// +kubebuilder:object:generate=false
```

Comment:
Just out of curiosity, what kind of object would kubebuilder generate for an interface?

Reply:
Nothing, it will print an error: 
```
github.com/grafana/grafana-operator/v5/api/v1beta1:-: invalid type: interface{AllowCrossNamespace() bool; MatchLabels() *k8s.io/apimachinery/pkg/apis/meta/v1.LabelSelector; MatchNamespace() string; ResyncPeriodHasElapsed() bool}
Error: not all generators ran successfully
run `controller-gen object:headerFile=hack/boilerplate.go.txt paths=./... -w` to see all available markers, or `controller-gen object:headerFile=hack/boilerplate.go.txt paths=./... -h` for usage
make: *** [Makefile:90: generate] Error 1
```
Which is the original reason I didn't have it defined under the `v1beta1` package.
Untill I found that specific annotation.